sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/sonar/TableWithoutHeaderCheck.java

```diff
@@ -37,12 +42,32 @@ private static boolean isTable(TagNode node) {
 
   private static boolean isLayout(TagNode node) {
     String role = node.getAttribute("role");
-    return role != null && ("PRESENTATION".equalsIgnoreCase(role) || "NONE".equalsIgnoreCase(role));
+    return "PRESENTATION".equalsIgnoreCase(role) || "NONE".equalsIgnoreCase(role);
   }
 
   private static boolean isHidden(TagNode node) {
     String ariaHidden = node.getAttribute("aria-hidden");
-    return ariaHidden != null && "TRUE".equalsIgnoreCase(ariaHidden);
+    return "TRUE".equalsIgnoreCase(ariaHidden);
+  }
+
+  private static boolean hasThymeleafFragmentInsertion(TagNode node) {
+    return hasThymeleafFragmentInsertionFromTableAttribute(node.getAttributes()) || hasThymeleafFragmentInsertionFromTableChildren(node.getChildren());
+  }
+
+  private static boolean hasThymeleafFragmentInsertionFromTableChildren(List<TagNode> nodes) {
+    boolean hasInsertion = false;
+    for (TagNode node : nodes) {
+      if (node.getNodeName().startsWith("th:block")) {
+        hasInsertion = hasInsertion || node.getAttributes().stream().map(Attribute::getName).anyMatch(THYMELEAF_FRAGMENT_INSERTION_KEYWORDS::contains);
+      } else {
+        hasInsertion = hasInsertion || node.getAttributes().stream().map(Attribute::getName).anyMatch(attributeName -> attributeName.equals("th:replace"));
+      }
+    }
+    return hasInsertion;
```

Comment:
Note that here we don't necessarily need to go through all the elements and have the `hasInsertion` boolean value be the incisive OR of all values: as soon as one of the condition is true, we could directly return true

Reply:
Yep, good catch. I've added a helper function, wrapped the conditions in one if, and I will return as soon as possible.